### PR TITLE
feat(): expose `values` object & `setValue` function on scope of defa…

### PR DESCRIPTION
**Pull request by carsoli**
…ult slot of KtFormControllerObject

setValue: a function that accepts `key`, `newValue` where `key` is an attribute on the values object of `KtFormControllerObject`
values: current value of `KtFormControllerObject`

### CODE REVIEW

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue [hunk 1]
```diff
@@ -41,19 +46,29 @@ export default defineComponent({
 			},
 		)
 
+		const setValue = (key, newValue) =>
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:49]: Why was this moved?

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue [hunk 2]
```diff
@@ -1,5 +1,10 @@
 <template>
-	<div><slot /></div>
+	<div>
+		<slot
+			:setValue="(key, newValue) => setValue(key, newValue)"
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:4]: Why isn’t this `setValues`?

File: packages/documentation/pages/components/form.vue [hunk 3]
```diff
@@ -271,6 +281,7 @@ export default defineComponent({
 			personalDetails: {
 				firstName: 'John',
 				lastName: 'Smith',
+				fullName: `Smith${Dashes.EmDash}John`,
```
Comment (FlorianWendelborn) [inline packages/documentation/pages/components/form.vue:284]: I think a hypen is more appropriate here.

File: packages/documentation/pages/components/form.vue [hunk 4]
```diff
@@ -81,18 +81,27 @@
...
+								label="First Name"
+								placeholder="Klaus"
+							/>
+							<KtFieldText
+								formKey="lastName"
+								helpText="help for lastName"
+								label="Last Name"
+								placeholder="Dieter"
+							/>
+							<KtFieldText
```
Comment (FlorianWendelborn) [inline packages/documentation/pages/components/form.vue:97]: I don’t have a better example in mind, but I think this doesn’t really fit the kind of intuitive example we wanted to have here

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue [hunk 5]
```diff
@@ -41,19 +46,29 @@ export default defineComponent({
...
-					...values.value,
-					[key]: newValue,
-				}),
+			setValue,
 			validators: context.validators,
 			values,
 		})
+
+		return {
+			/**
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:66]: JSDoc strings should be attached to the definition

File: packages/documentation/pages/components/form.vue [hunk 6]
```diff
@@ -81,18 +81,27 @@
...
-						<KtFieldText
-							formKey="lastName"
-							helpText="help for lastName"
-							label="Last Name"
-							placeholder="Dieter"
-						/>
+						<template #default="{ setValue, values }">
+							<KtFieldText
+								formKey="firstName"
+								helpText="help for lastName"
```
Comment (FlorianWendelborn) [inline packages/documentation/pages/components/form.vue:87]: might as well fix that one kek

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue (hunk 2 above)
Comment (carsoli) [reply packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:4]: sets a key only

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue (hunk 1 above)
Comment (carsoli) [reply packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:49]: needed in the return statement

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue (hunk 2 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:4]: Yes, but why isn’t `setValues` exposed? It just makes sense to have `x` and `setX` exposed

File: packages/documentation/pages/components/form.vue (hunk 3 above)
Comment (carsoli) [reply packages/documentation/pages/components/form.vue:284]: ok...

File: packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue [hunk 7]
```diff
@@ -41,19 +47,33 @@ export default defineComponent({
...
-				}),
+			setValue,
 			validators: context.validators,
 			values,
 		})
+
+		return {
+			setValue,
+			/**
+			 * Updates a single key in a KtFormControllerObject values
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:72]: I think this docstring is wrong

Comment (carsoli) [reply packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:72]: can u elaborate

Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:72]: > single key
> `setValues`

Comment (carsoli) [reply packages/kotti-ui/source/kotti-form/controllers/KtFormControllerObject.vue:72]: copy pasta mistake, fixing

**Review by FlorianWendelborn — APPROVED** [review]
